### PR TITLE
Domains: Update message on incoming transfer pre-check if privacy protection is detected

### PR DESCRIPTION
**Pull request by danhauk**
This updates the messaging on the incoming transfer pre-check screen if privacy protection is detected. We need to make sure the user has privacy turned off before they try to start the transfer.

Before | After
------------ | -------------
image | image

#### Testing

1. Start at `[/domains/add/transfer/](http://calypso.localhost:3000/domains/add/transfer/)`
2. Select a site to transfer a domain
3. Enter a domain that has a known privacy contact address (any addresses with `private`, `privacy`, `domainsbyproxy`, `whoisproxy`, or `whoisguard`)
4. Confirm it looks like the **After** screenshot above
5. Go back and enter a domain that doesn't have privacy on it
6. Confirm it looks like the **Before** screenshot above

### CODE REVIEW

**Comment by danhauk** [comment]
@delputnam How reliable is the check for privacy on the backend? If the email address contains one of those strings it will show as privacy enabled, correct? The reason I ask is because I was considering blocking the user from proceeding to step 3 if we detect privacy. But if that could potentially get someone stuck even if privacy is off we shouldn't do that.

Also, cc @ranh a copy review on this would be great.

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 1]
```diff
@@ -330,6 +330,34 @@ class TransferDomainPrecheck extends React.PureComponent {
 			buttonText = translate( 'I can access the email address listed' );
 		}
 
+		if ( privacy ) {
+			message = translate(
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: You can probably just extract the notice. :)

Comment (aidvu) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Actually, nm. Probably not reusable.

**Comment by aidvu** [comment]
@danhauk the backend checks for specific strings in the admin e-mail (if we can get it). You could say it's reliable. I mean, once we figure it's non-forwarding, we add a string in there. It should get better with time.

**Comment by danhauk** [comment]
@aidvu Yes, I guess I should've phrased that better. It's reliable in that it checks for strings, but it could block the user if their actual admin email is something like `domain.admin+private@gmail.com` and we show a false positive right?

**Comment by aidvu** [comment]
@danhauk it would, yes. :) Custom e-mails are a problem, tho usually these have fixed domains, like `domainsbyproxy.com` or similar. I'd also like to hear what @delputnam thinks. :)

**Comment by wensco** [comment]
A+ for stronger warning notices here - nice! Just a couple of comments...

Let's not block the user if we detect privacy. For example, my registrar (shown in the screenshots) has email forwarding which means that I can leave privacy active and still transfer. We may also have a case as you noted where a user's email contains a word that makes us think they have privacy but they don't.

In that spirit, we may also want to adjust the warning message to be a little less...assertive. :)

Maybe something like "It looks like you have privacy protection enabled...." since it's possible they don't and we just think they do.

**Comment by delputnam** [comment]
When privacy is `true`, it means that either, 1) we couldn't get any `admin_email` info (either because something failed or because the registry didn't provide it). or 2) the string we got contains one of those 6 strings in the privacy pattern list.

Since we don't have a comprehensive list of all of the email domains used by the privacy providers, using a search for strings like `private` or `privacy` seems to catch a lot of them, but as you say, it could generate false positives.

I think that right now, it's probably better to allow the user to bypass the step, but this is definitely something to revisit as we get better at this.

**Pull request merged**